src/tracer/MARBL_tracers.F90

```diff
@@ -0,0 +1,734 @@
+
+! Local variables
+  character(len=256) :: log_message
+  real, dimension(SZI_(G),SZJ_(G),SZK_(G)) :: h_work ! Used so that h can be modified
+  real :: sfc_val  ! The surface value for the tracers.
+  real :: Isecs_per_year  ! The number of seconds in a year.
+  real :: year            ! The time in years.
+  integer :: secs, days   ! Integer components of the time type.
+  integer :: i, j, k, is, ie, js, je, nz, m
+
+  is = G%isc ; ie = G%iec ; js = G%jsc ; je = G%jec ; nz = GV%ke
+
+  if (.not.associated(CS)) return
+  if (CS%ntr < 1) return
+
+  ! (1) Compute surface fluxes
+  ! FIXME: MARBL can handle computing surface fluxes for all columns simultaneously
+  !        I was just thinking going column-by-column at first might be easier
+  do i=is,ie
+    do j=js,je
+      ! (1) Load proper column data
+      !     i. surface flux forcings
+
+      !        These fields are getting the correct data
+      if (CS%sss_ind > 0) marbl_instances%surface_flux_forcings(CS%sss_ind)%field_0d(1) = tv%S(i,j,1)
+      if (CS%sst_ind > 0) marbl_instances%surface_flux_forcings(CS%sst_ind)%field_0d(1) = tv%T(i,j,1)
+      if (CS%ifrac_ind > 0) marbl_instances%surface_flux_forcings(CS%ifrac_ind)%field_0d(1) = fluxes%ice_fraction(i,j)
+      ! MARBL wants u10_sqr in (cm/s)^2
+      if (CS%u10_sqr_ind > 0) marbl_instances%surface_flux_forcings(CS%u10_sqr_ind)%field_0d(1) = fluxes%u10_sqr(i,j) * (US%L_t_to_m_s * cm_per_m)**2
+      ! mct_driver/ocn_cap_methods:93 -- ice_ocean_boundary%p(i,j) comes from coupler
+      ! We may need a new ice_ocean_boundary%p_atm because %p includes ice in GFDL driver
+      if (CS%atmpress_ind > 0) marbl_instances%surface_flux_forcings(CS%atmpress_ind)%field_0d(1) = fluxes%p_surf_full(i,j) * US%R_to_kg_m3 * US%L_T_to_m_s**2 * atm_per_Pa
+
+      ! These are okay, but need option to come in from coupler
+      if (CS%xco2_ind > 0) marbl_instances%surface_flux_forcings(CS%xco2_ind)%field_0d(1) = CS%atm_co2_const
+      if (CS%xco2_alt_ind > 0) marbl_instances%surface_flux_forcings(CS%xco2_alt_ind)%field_0d(1) = CS%atm_alt_co2_const
+
+      !        These fields are receiving physically-reasonable values
+      if (CS%dust_dep_ind > 0) marbl_instances%surface_flux_forcings(CS%dust_dep_ind)%field_0d(1) = 0 ! pop gets from coupler
+      if (CS%fe_dep_ind > 0) marbl_instances%surface_flux_forcings(CS%fe_dep_ind)%field_0d(1) = 0 ! pop gets from coupler (but can read from file)
+
+      !        Read these from /glade/work/mlevy/cesm_inputdata/ndep_ocn_1850_w_nhx_emis_MOM_tx0.66v1_c200827.nc
+      ! MOM_read_data()
+      if (CS%nox_flux_ind > 0) marbl_instances%surface_flux_forcings(CS%nox_flux_ind)%field_0d(1) = 0 ! pop gets from file
+      if (CS%nhy_flux_ind > 0) marbl_instances%surface_flux_forcings(CS%nhy_flux_ind)%field_0d(1) = 0 ! pop gets from file
+
+      !     ii. tracers at surface
+      do m=1,CS%ntr
+        marbl_instances%tracers_at_surface(1,m) = CS%tr(i,j,1,m)
```

Comment:
Are you intending to be able to modify these? If so might want to make this a pointer instead

Reply:
Nope, MARBL won't modify these; `marbl_instances%tracers_at_surface` is passed as an `intent(in)` from `marbl_instances%surface_flux_compute()`